integration-tests/test_redaction.py

```diff
@@ -0,0 +1,131 @@
+import pytest
+import subprocess
+import tarfile
+import os
+import yaml
+import uuid
+from enum import Enum
+
+pytestmark = pytest.mark.usefixtures("register_subman")
+TEST_CMD = ["/usr/bin/uptime", "/sbin/lsmod", "/bin/ls", "/sbin/ethtool"]
+TEST_FILE = ["/proc/cpuinfo", "/proc/meminfo", "/etc/yum.conf", "/etc/os-release"]
+FILE_REDACTION_FILE = "/etc/insights-client/file-redaction.yaml"
+CONTENT_REDACTION_FILE = "/etc/insights-client/file-content-redaction.yaml"
+
+
+@pytest.mark.parametrize("not_removed_command", TEST_CMD)
+def test_redaction_not_on_cmd(insights_client, tmp_path, not_removed_command):
+    """
+    Do not configure commands in /etc/insights-client/file-redaction.yaml,
+    so related commands should exist in collection archive.
+    """
+    assert check_archive(insights_client, tmp_path, not_removed_command)
+
+
+@pytest.mark.parametrize("removed_command", TEST_CMD)
+def test_redaction_on_cmd(insights_client, tmp_path, removed_command):
+    """
+    Configure command redaction in /etc/insights-client/file-redaction.yaml,
+    so related commands should not exist in collection archive.
+    """
+    try:
+        configure_file_redaction("Command", removed_command)
+        assert not check_archive(insights_client, tmp_path, removed_command)
+    finally:
+        os.remove(FILE_REDACTION_FILE)
+
+
+@pytest.mark.parametrize("not_removed_file", TEST_FILE)
+def test_redaction_not_on_file(insights_client, tmp_path, not_removed_file):
+    """
+    Do not configure files in /etc/insights-client/file-redaction.yaml,
+    so related files should exist in collection archive.
+    """
+    assert check_archive(insights_client, tmp_path, not_removed_file)
+
+
+@pytest.mark.parametrize("removed_file", TEST_FILE)
+def test_redaction_on_file(insights_client, tmp_path, removed_file):
+    """
+    Redact files in /etc/insights-client/file-redaction.yaml,
+    so related files should not exist in collection archive.
+    """
+    try:
+        configure_file_redaction("File", removed_file)
+        assert not check_archive(insights_client, tmp_path, removed_file)
+    finally:
+        os.remove(FILE_REDACTION_FILE)
+
+
+def test_redaction_on_pattern_hostname(insights_client, tmp_path):
+    """
+    Test if the satellite hostname is obfuscated
+    Related to https://issues.redhat.com/browse/RHEL-2471
+
+    """
+
+    # Record the current hostname
+    with open("/etc/hostname", "r") as f:
+        previous_hostname = f.read().strip()
+
+    try:
+        # Set a new hostname
+        hostname_to_test = "insights-client-satellite"
```

Comment:
can we make it generic for hosted and satellite by removing 'satellite' word from here?
I think this test applies to both configurations.

Reply:
Done